Dockerfile

```diff
@@ -181,6 +193,10 @@ COPY --from=async-profiler-builder-glibc /tmp/async-profiler/build/fdtransfer gp
 
 COPY --from=rbspy-builder /tmp/rbspy/rbspy gprofiler/resources/ruby/rbspy
 
+COPY --from=dotnet-builder /usr/share/dotnet/host /usr/share/dotnet/host
```

Comment:
What are we copying from these 2 directories? i.e `gprofiler/resources/dotnet` is the `dotnet-trace` app itself, and what are these 2 directories?

Reply:
These directories contain libraries needed to run the dotnet-trace tool. I determined them by brute force - removing the directories from the /usr/share/dotnet and checking whether it would still work. It reduced the size by more than 2/3 and I hope that's ok.

Reply:
Which libraries precisely are required? Can it be determined with `ldd`? (i.e `ldd dotnet-trace`?)

Reply:
ldd output did not really help:
`        linux-vdso.so.1 (0x00007ffdc2cb4000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6d79aa0000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f6d7989c000)
        libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f6d79513000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6d79175000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6d78f5d000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6d78b6c000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6d79cbf000)`
        
I determined dotnet-trace libraries manually by moving each one and checking if it still works. Here they are:

`3.3M    dotnet/shared/Microsoft.NETCore.App/6.0.8/libclrjit.so 
7.1M    dotnet/shared/Microsoft.NETCore.App/6.0.8/libcoreclr.so
376K    dotnet/shared/Microsoft.NETCore.App/6.0.8/libhostpolicy.so
92K     dotnet/shared/Microsoft.NETCore.App/6.0.8/libSystem.Native.so
812K    dotnet/shared/Microsoft.NETCore.App/6.0.8/Microsoft.CSharp.dll
60K     dotnet/shared/Microsoft.NETCore.App/6.0.8/Microsoft.NETCore.App.deps.json
16K     dotnet/shared/Microsoft.NETCore.App/6.0.8/Microsoft.Win32.Primitives.dll
48K     dotnet/shared/Microsoft.NETCore.App/6.0.8/mscorlib.dll
92K     dotnet/shared/Microsoft.NETCore.App/6.0.8/netstandard.dll
224K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Collections.Concurrent.dll
256K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Collections.dll
8.0K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.ComponentModel.dll
64K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.ComponentModel.Primitives.dll
176K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Console.dll
16K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Core.dll
2.9M    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Data.Common.dll
8.0K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Data.DataSetExtensions.dll
16K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Data.dll
252K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Diagnostics.Process.dll
40K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.dll
16K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.IO.FileSystem.dll
520K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Linq.dll
3.8M    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Linq.Expressions.dll
164K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Memory.dll
10M     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Private.CoreLib.dll
32K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Runtime.dll
516K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Text.RegularExpressions.dll
116K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Threading.Channels.dll
72K     dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Threading.dll
8.0K    dotnet/shared/Microsoft.NETCore.App/6.0.8/System.Threading.Overlapped.dll`

It is 31M in total, and 37M can be removed. Is it ok? Hopefully there's a way to copy all these dependencies as a list determined elsewhere, otherwise the Dockerfile would look atrocius

Reply:
Yes, that's fine, let's copy just them like you did :)